### PR TITLE
Simple login/logout logic with Context API

### PR DESCRIPTION
**Pull request by mushegh-aob**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -28,16 +27,17 @@ const App = props => {
         path="/"
         withAuth={true}
         redirectOnFailure="/login"
-        isAuthenticated={isAuthenticated}
-        render={() => (
-          <Switch>
-            {routes.map((route, index) => (
-              <Route key={index} {...route} />
-            ))}
+        render={() => {
```
Comment (hpilosyan) [inline src/App.js]: Why did you need this change?

File: src/App.js [hunk 2]
```diff
@@ -4,8 +4,8 @@ import AuthenticationRoute from "./containers/AuthenticationRoute";
 import routes from "./helpers/routes";
 import LoginScreen from "./components/LoginScreen";
 
-const App = props => {
-  const isAuthenticated = false;
+const App = ({ location }) => {
+  const { from } = location.state || "home";
```
Comment (hpilosyan) [inline src/App.js]: Why?

File: src/hooks/useAuth.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import React, { useContext, useState, useEffect } from "react";
```
Comment (hpilosyan) [inline src/hooks/useAuth.js]: Lets discuss this, so that I have better understanding what I review

File: src/components/HomeScreen/index.js [hunk 4]
```diff
@@ -1,13 +1,20 @@
-import React from "react";
+import React, { useContext } from "react";
+import AuthContext from "./../../context/AuthContext";
 import { useTranslation } from "react-i18next";
 import LanguageContainer from "./../LanguageContainer";
 import styles from "./index.module.scss";
 
-const HomeScreen = props => {
+const HomeScreen = () => {
+  const { user, logout } = useContext(AuthContext);
```
Comment (hpilosyan) [inline src/components/HomeScreen/index.js]: This is not right, you should get user through props

File: src/containers/AuthenticationRoute/index.js [hunk 5]
```diff
@@ -1,26 +1,40 @@
 import React from "react";
 import { Route, Redirect, withRouter } from "react-router-dom";
+import AuthContext from "./../../context/AuthContext";
 
 class AuthenticationRoute extends Route {
+  // Assign a contextType to get isAuthed from AuthContext
+  static contextType = AuthContext;
```
Comment (hpilosyan) [inline src/containers/AuthenticationRoute/index.js]: Again you shouldn't import context directly

File: src/context/AuthContext.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+import { TOKEN } from "./../constants";
+
+const initialState = {
+  isAuthed: false,
+  authenticate: () => {},
+  logout: () => {},
+  user: null
+};
+
+const AuthContext = createContext(initialState);
```
Comment (hpilosyan) [inline src/context/AuthContext.js]: Move this into useAuth hook and don't export it, so that you're not tempted to import it in other places :)

File: src/context/AuthContext.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import { createContext } from "react";
```
Comment (hpilosyan) [inline src/context/AuthContext.js]: This file name shouldn't be AuthContext.js , but rather auth.js or someting like that

File: src/hooks/useAuth.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+import AuthContext, {
```
Comment (hpilosyan) [inline src/hooks/useAuth.js]: AuthContext shouldn't be upper case

File: src/containers/AuthenticationRoute/index.js (hunk 5 above)
Comment (mushegh-aob) [reply src/containers/AuthenticationRoute/index.js]: Hooks don't work inside classes. Don't know other way.

File: src/hooks/useAuth.js (hunk 8 above)
Comment (mushegh-aob) [reply src/hooks/useAuth.js]: do You mean it should look  like this ? : 
`const authContext = createContext(null);

<authContext.Provider
      value={{
        isAuthed,
        authenticate,
        logout,
        user
      }}
    >
      {children}
    </authContext.Provider>
`

**Comment by hpilosyan** [comment]
Please rebase

File: src/containers/AuthenticationRoute/index.js (hunk 5 above)
Comment (mushegh-aob) [reply src/containers/AuthenticationRoute/index.js]: Instead of static contextType, now  I use AuthContext.Consumer here to grab isAuthed
